myTest.java

```diff
@@ -0,0 +1,177 @@
+import com.google.gson.Gson;
+import org.junit.jupiter.api.*;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import parser.Parser;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import static org.junit.jupiter.api.Assertions.*;
+
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class myTest {
+    private Cart ilyaCart;
+
+    private RealItem car;
+    private String realItemCar;
+
+    private VirtualItem game;
+    private String virtualItemGame;
+
+    private String itemsAll;
+
+    private Double expectedPrice;
+    private static final double TAX = 0.2;
+
+    private Parser parser = new JsonParser();
+
+    private Gson gson = new Gson();
+
+    private String ilyaCartObjectContent;
+
+    private final ByteArrayOutputStream outShowItemsContent = new ByteArrayOutputStream();
+    private final PrintStream originalOut = System.out;
+
+    @AfterAll
+    public static void deleteFile() {
+        File ilyaCartFile = new File("src/main/resources/ilya-cart.json");
+        ilyaCartFile.deleteOnExit();
+    }
+
+    @AfterEach
+    public void setOriginalSystemOut(){
+        System.setOut(originalOut);
+    }
+
+    @BeforeEach
+    public void setUP()  {
+        ilyaCart = new Cart("ilya-cart");
+
+        car = new RealItem();
+        car.setName("KIA");
+        car.setPrice(20000.0);
+        car.setWeight(1600.0);
+        realItemCar  = "Class: class shop.RealItem; Name: KIA; Price: 20000.0; Weight: 1600.0";
+
+        game = new VirtualItem();
+        game.setName("Star Wars");
+        game.setPrice(100);
+        game.setSizeOnDisk(36000.0);
+        virtualItemGame = "Class: class shop.VirtualItem; Name: Star Wars; Price: 100.0; Size on disk: 36000.0";
+
+        itemsAll = realItemCar + "\r\n" + virtualItemGame + "\r\n";
+
+        ilyaCart.addRealItem(car);
+        ilyaCart.addVirtualItem(game);
+
+        expectedPrice = game.getPrice() + game.getPrice()*TAX + car.getPrice() + car.getPrice()*TAX;
+
+        ilyaCartObjectContent = gson.toJson(ilyaCart);
+
+        System.setOut(new PrintStream(outShowItemsContent));
+    }
+
+    @Test
+    @Order(3)
+    @Tag("Cart")
+    public void testCartName() {
+        assertEquals("ilya-cart", ilyaCart.getCartName());

```

Comment:
We try to follow AAA pattern (http://wiki.c2.com/?ArrangeActAssert) for all tests. Let's make all of them in such pattern.

Reply:
I've made some changes to support the AAA pattern:
1) Separate the data inside the test
2) I've got tests, where data set up in the beginning:
for this case, I moved everything that I can from "setup" into the tests. (everything which is not used for other tests).
The problem that I wasn't solved yet:
I've got a setup BeforeEach, where I set values that will be used by different tests. This data isn't necessary for all tests. But it is used in most of them! And it would be incorrect if some tests will contain the same code(testing data setup).

Reply:
You're correct - common part can be in before hook. But sometimes it even make sense to repeat some part in test to have good readability.

Reply:
I suppose this conversation can be resolved.